bigquery/snippets/snippet.go

```diff
@@ -142,10 +141,86 @@ func createTable(client *bigquery.Client, datasetID, tableID string) error {
 	if err := table.Create(ctx, &bigquery.TableMetadata{Schema: schema}); err != nil {
 		return err
 	}
+	return nil
+}
+
+func createTableExplicitSchema(client *bigquery.Client, datasetID, tableID string) error {
+	ctx := context.Background()
+	// [START bigquery_create_table]
+	// Represent email_addrs as an array of strings, rather than a single address
+	sampleSchema := bigquery.Schema{
+		{Name: "full_name", Type: bigquery.StringFieldType},
+		{Name: "age", Type: bigquery.IntegerFieldType},
+	}
+
+	metaData := &bigquery.TableMetadata{
+		Schema:         sampleSchema,
+		ExpirationTime: time.Now().AddDate(1, 0, 0), // Table will be automatically deleted in 1 year
+	}
+	tableRef := client.Dataset(datasetID).Table(tableID)
+	if err := tableRef.Create(ctx, metaData); err != nil {
+		return err
+	}
 	// [END bigquery_create_table]
 	return nil
 }
 
+func createTableEmptySchema(client *bigquery.Client, datasetID, tableID string) error {
+	ctx := context.Background()
+	// [START bigquery_create_table_without_schema]
+	// Create the table as a partitioned table
+	metaData := &bigquery.TableMetadata{
+		TimePartitioning: &bigquery.TimePartitioning{
+			Expiration: time.Duration(24*365) * time.Hour, // 365 day partition expiry
+		},
+	}
+	if err := client.Dataset(datasetID).Table(tableID).Create(ctx, metaData); err != nil {
+		return err
+	}
+	// [END bigquery_create_table_without_schema]
+	return nil
+}
+
+func updateTableDescription(client *bigquery.Client, datasetID, tableID string) error {
+	ctx := context.Background()
+	// [START bigquery_update_table_description]
+	tableRef := client.Dataset(datasetID).Table(tableID)
+	original, err := tableRef.Metadata(ctx)
+	if err != nil {
+		return err
+	}
+	newMeta := bigquery.TableMetadataToUpdate{
+		Description: "Updated description.", // table expiration in 5 days
```

Comment:
`// table expiration in 5 days` copypasta?

Reply:
Its gluten free! Removed.